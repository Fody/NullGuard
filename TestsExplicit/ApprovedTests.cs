#if(DEBUG)

using System.Linq;
using ApprovalTests;
using NUnit.Framework;

[TestFixture]
public class ApprovedTests
{
    [Test]
    public void ClassWithBadAttributes()
    {
        Approvals.Verify(Decompiler.Decompile(AssemblyWeaver.AfterAssemblyPath, "ClassWithBadAttributes"));
    }

    [Test]
    public void ClassWithPrivateMethod()
    {
        Approvals.Verify(Decompiler.Decompile(AssemblyWeaver.AfterAssemblyPath, "ClassWithPrivateMethod"));
    }

    [Test]
    public void ClassWithPrivateMethodNoAssert()
    {
        Approvals.Verify(Decompiler.Decompile(AssemblyWeaver.AfterAssemblyPath, "ClassWithPrivateMethod"));
    }

    [Test]
    public void GenericClass()
    {
        Approvals.Verify(Decompiler.Decompile(AssemblyWeaver.AfterAssemblyPath, "GenericClass`1"));
    }

    [Test]
    public void Indexers()
    {
        Approvals.Verify(Decompiler.Decompile(AssemblyWeaver.AfterAssemblyPath, "Indexers"));
    }

    [Test]
    public void InterfaceBadAttributes()
    {
        Approvals.Verify(Decompiler.Decompile(AssemblyWeaver.AfterAssemblyPath, "InterfaceBadAttributes"));
    }

    [Test]
    public void SimpleClass()
    {
        Approvals.Verify(Decompiler.Decompile(AssemblyWeaver.AfterAssemblyPath, "SimpleClass"));
    }

    [Test]
    public void SimpleClassNoAssert()
    {
        Approvals.Verify(Decompiler.Decompile(AssemblyWeaver.AfterAssemblyPath, "SimpleClass"));
    }

    [Test]
    public void SkipIXamlMetadataProvider()
    {
        Approvals.Verify(Decompiler.Decompile(AssemblyWeaver.AfterAssemblyPath, "XamlMetadataProvider"));
    }

    [Test]
    public void SpecialClass()
    {
        Approvals.Verify(Decompiler.Decompile(AssemblyWeaver.AfterAssemblyPath, "SpecialClass"));
    }

    [Test]
    public void PublicNestedInsideNonPublic()
    {
        Approvals.Verify(Decompiler.Decompile(AssemblyWeaver.AfterAssemblyPath, "NonPublicWithNested"));
    }

    [Test]
    public void UnsafeClass()
    {
        Approvals.Verify(Decompiler.Decompile(AssemblyWeaver.AfterAssemblyPath, "UnsafeClass"));
    }

    [Test]
    public void DerivedClass()
    {
        Approvals.Verify(Decompiler.Decompile(AssemblyWeaver.AfterAssemblyPath, "DerivedClass"));
    }

    [Test]
    public void ImplementsInterface()
    {
        Approvals.Verify(Decompiler.Decompile(AssemblyWeaver.AfterAssemblyPath, "ImplementsInterface"));
    }

    [Test]
    public void ImplementsInheritedInterface()
    {
        Approvals.Verify(Decompiler.Decompile(AssemblyWeaver.AfterAssemblyPath, "ImplementsInheritedInterface"));
    }

    [Test]
    public void ImplementsInterfaceExplicit()
    {
        Approvals.Verify(Decompiler.Decompile(AssemblyWeaver.AfterAssemblyPath, "ImplementsInterfaceExplicit"));
    }

    [Test]
    public void InfosList()
    {
        Approvals.VerifyAll(AssemblyWeaver.Infos.OrderBy(e => e), "Infos: ");
    }

    [Test]
    public void WarnsList()
    {
        Approvals.VerifyAll(AssemblyWeaver.Warns.OrderBy(e => e), "Warns: ");
    }

    [Test]
    public void ErrorsList()
    {
        Approvals.VerifyAll(AssemblyWeaver.Errors.OrderBy(e => e), "Errors: ");
    }
}

#endif