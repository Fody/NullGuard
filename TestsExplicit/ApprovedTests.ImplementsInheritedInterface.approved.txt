.class public auto ansi beforefieldinit ImplementsInheritedInterface
       extends [mscorlib]System.Object
       implements InheritedInterface,
                  BaseInterfaceWithAttributes
{
  .field private string '<NotNullProperty>k__BackingField'
  .method public hidebysig newslot virtual final 
          instance void  MethodWithNotNullParameter(string canBeNull,
                                                    string arg) cil managed
  {
    // Code size       20 (0x14)
    .maxstack  2
    .language '{3F5162F8-07C6-11D3-9053-00C04FA302A1}', '{994B45C4-E6E9-11D2-903F-00C04FA302A1}', '{5A869D0B-6611-11D3-BD2A-0000F80849BD}'
    IL_0000:  ldarg.2
    IL_0001:  brtrue.s   IL_0013
    IL_0003:  ldstr      "arg"
    IL_0008:  ldstr      "[NullGuard] arg is null."
    IL_000d:  newobj     instance void [mscorlib]System.ArgumentNullException::.ctor(string,
                                                                                     string)
    IL_0012:  throw
    IL_0013:  ret
  } 
  .method public hidebysig newslot virtual final 
          instance string  MethodWithNotNullReturnValue(string arg) cil managed
  {
    // Code size       17 (0x11)
    .maxstack  2
    IL_0000:  ldarg.1
    IL_0001:  dup
    IL_0002:  brtrue.s   IL_0010
    IL_0004:  pop
    IL_0005:  ldstr      "[NullGuard] Return value of method 'System.String "
    + "ImplementsInheritedInterface::MethodWithNotNullReturnValue(System.Strin"
    + "g)' is null."
    IL_000a:  newobj     instance void [mscorlib]System.InvalidOperationException::.ctor(string)
    IL_000f:  throw
    IL_0010:  ret
  } 
  .method public hidebysig newslot specialname virtual final 
          instance string  get_NotNullProperty() cil managed
  {
    // Code size       22 (0x16)
    .maxstack  2
    IL_0000:  ldarg.0
    IL_0001:  ldfld      string ImplementsInheritedInterface::'<NotNullProperty>k__BackingField'
    IL_0006:  dup
    IL_0007:  brtrue.s   IL_0015
    IL_0009:  pop
    IL_000a:  ldstr      "[NullGuard] Return value of property 'System.Strin"
    + "g ImplementsInheritedInterface::NotNullProperty()' is null."
    IL_000f:  newobj     instance void [mscorlib]System.InvalidOperationException::.ctor(string)
    IL_0014:  throw
    IL_0015:  ret
  } 
  .method public hidebysig newslot specialname virtual final 
          instance void  set_NotNullProperty(string 'value') cil managed
  {
    // Code size       27 (0x1b)
    .maxstack  2
    IL_0000:  ldarg.1
    IL_0001:  brtrue.s   IL_0013
    IL_0003:  ldstr      "value"
    IL_0008:  ldstr      "[NullGuard] Cannot set the value of property 'Syst"
    + "em.String ImplementsInheritedInterface::NotNullProperty()' to null."
    IL_000d:  newobj     instance void [mscorlib]System.ArgumentNullException::.ctor(string,
                                                                                     string)
    IL_0012:  throw
    IL_0013:  ldarg.0
    IL_0014:  ldarg.1
    IL_0015:  stfld      string ImplementsInheritedInterface::'<NotNullProperty>k__BackingField'
    IL_001a:  ret
  } 
  .method public hidebysig specialname rtspecialname 
          instance void  .ctor() cil managed
  {
    // Code size       7 (0x7)
    .maxstack  1
    IL_0000:  ldarg.0
    IL_0001:  call       instance void [mscorlib]System.Object::.ctor()
    IL_0006:  ret
  } 
  .property instance string NotNullProperty()
  {
  } 
}
